### PR TITLE
Evaluate background metric variables on FD grids

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

Add a new mutator for initializing and updating background metric variables on cell-centered and face-centered FD grids. This is a follow-up work of #5028.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/Evolution/DgSubcell/Test_BackgroundGrVars.cpp [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+      tmpl::list<gr::Tags::Lapse<DataVector>, gr::Tags::Shift<DataVector, 3>>>;
+  using flux_spacetime_variables_tag =
+      ::Tags::Variables<tmpl::list<gr::Tags::SqrtDetSpatialMetric<DataVector>,
+                                   gr::Tags::SpatialMetric<DataVector, 3>>>;
+  using inverse_spatial_metric_tag =
+      gr::Tags::InverseSpatialMetric<DataVector, 3>;
+};
+
+// A free function returning a moving brick domain
+template <bool mesh_is_moving>
```
Comment (wthrowe) [inline tests/Unit/Evolution/DgSubcell/Test_BackgroundGrVars.cpp]: `MeshIsMoving`

File: tests/Unit/Evolution/DgSubcell/Test_BackgroundGrVars.cpp [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+    const Mesh<3> face_centered_mesh{extents, basis, quadrature};
+    gsl::at(face_centered_meshes, dim) = Mesh<3>{extents, basis, quadrature};
+  }
+  return face_centered_meshes;
+}
+
+// if `test_runtime_initial_data` == false, test for the compile time initial
+// data
+template <bool test_moving_mesh, bool test_runtime_initial_data,
+          bool compute_only_on_rollback>
```
Comment (wthrowe) [inline tests/Unit/Evolution/DgSubcell/Test_BackgroundGrVars.cpp]: Camel-case all of these.

File: tests/Unit/Evolution/DgSubcell/Test_BackgroundGrVars.cpp [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+        check_face_centered_vars(expected_initial_face_centered_gr_vars);
+      } else {
+        check_cell_centered_vars(expected_later_cell_centered_gr_vars, true);
+        check_face_centered_vars(expected_later_face_centered_gr_vars);
+      }
+    }
+  }
+}
+
+template <bool test_moving_mesh, bool test_runtime_initial_data>
```
Comment (wthrowe) [inline tests/Unit/Evolution/DgSubcell/Test_BackgroundGrVars.cpp]: Camel-case again.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by yoonso0-0** [comment]
Pushed the fixup. Also pushed another commit making same changes to the DG code.

**Comment by wthrowe** [comment]
Thanks for fixing that one too.  Go ahead and squash.

**Review by wthrowe — APPROVED** [review]

**Pull request merged**